rbd/encryption.go

```diff
@@ -0,0 +1,117 @@
+// +build !octopus,!nautilus
+
+package rbd
+
+// #cgo LDFLAGS: -lrbd
+// /* force XSI-complaint strerror_r() */
+// #define _POSIX_C_SOURCE 200112L
+// #undef _GNU_SOURCE
+// #include <errno.h>
+// #include <stdlib.h>
+// #include <rados/librados.h>
+// #include <rbd/librbd.h>
+import "C"
+
+import (
+	"unsafe"
+)
+
+// formt is the encryption format type
+type format C.rbd_encryption_format_t
+
+// Possible values for format:
+// EncryptionFormatLUKS1: LUKS v1
+// EncryptionFormatLUKS2: LUKS v2
+const (
+	EncryptionFormatLUKS1 = format(C.RBD_ENCRYPTION_FORMAT_LUKS1)
+	EncryptionFormatLUKS2 = format(C.RBD_ENCRYPTION_FORMAT_LUKS2)
+)
+
+// EncryptionAlgorithm is the encryption algorithm
+type EncryptionAlgorithm C.rbd_encryption_algorithm_t
+
+// Possible values for EncryptionAlgorithm:
+// EncryptionAlgorithmAES128: AES 128bits
+// EncryptionAlgorithmAES256: AES 256bits
+const (
+	EncryptionAlgorithmAES128 = EncryptionAlgorithm(C.RBD_ENCRYPTION_ALGORITHM_AES128)
+	EncryptionAlgorithmAES256 = EncryptionAlgorithm(C.RBD_ENCRYPTION_ALGORITHM_AES256)
+)
+
+// EncryptionFormatOptsLUKS1 and EncryptionFormatOptsLUKS2 are identical
+// structures at the moment, just as they are in the librbd api.
+// The purpose behind creating different identical structures, is to facilitate
+// future modifications of one of the formats, while maintaining backwards
+// compatibility with the other.
+
+// EncryptionFormatOptsLUKS1 options required for LUKS v1
+type EncryptionFormatOptsLUKS1 struct {
+	Alg        EncryptionAlgorithm
+	Passphrase []byte
+}
+
+// EncryptionFormatOptsLUKS2 options required for LUKS v2
+type EncryptionFormatOptsLUKS2 struct {
+	Alg        EncryptionAlgorithm
+	Passphrase []byte
+}
```

Comment:
In analogy to `EncrytionAlgorithm` the structs for the `EncryptionOptions` type should be called `EncryptionOptionsLUKSx`